[BE] Speed up runtime of test_ddp_model_diff_across_ranks

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55659 [BE] Speed up runtime of test_ddp_model_diff_across_ranks**

As per https://github.com/pytorch/pytorch/issues/55583, this is the most expensive distributed test.

Instead of waiting for process 0 in this test to be taken down by
nccl_async_error_handling, just remove the barrier and let the process exit
when the backend is NCCL.

A slight downside here is that the test no longer verifies that the process
would be brought down by nccl_async_error_handling, but
nccl_async_error_handling is already well tested in other tests. If we feel we
need to ensure this for this test, then we can pass in a process group with a
smaller timeout as an alternative solution.

The test now runs in 8-9s as opposed to 70. Ran the test 1000 times to verify
no flakiness

Differential Revision: [D27672161](https://our.internmc.facebook.com/intern/diff/D27672161/)

rohan-varma committed:
[BE] Speed up runtime of test_ddp_model_diff_across_ranks

As per https://github.com/pytorch/pytorch/issues/55583, this is the most expensive distributed test.

Instead of waiting for process 0 in this test to be taken down by
nccl_async_error_handling, just remove the barrier and let the process exit
when the backend is NCCL.

A slight downside here is that the test no longer verifies that the process
would be brought down by nccl_async_error_handling, but
nccl_async_error_handling is already well tested in other tests. If we feel we
need to ensure this for this test, then we can pass in a process group with a
smaller timeout as an alternative solution.

The test now runs in 4-6s as opposed to 70. Ran the test 1000 times to verify
no flakiness

Differential Revision: [D27672161](https://our.internmc.facebook.com/intern/diff/D27672161/)

[ghstack-poisoned]

mrshenli reviewed (approved):
This change LGTM. Stamp to unblock

Curious, is it possible to reduce the response time of `NCCL_ASYNC_ERROR_HANDLING` for just this test? I would assume there is a timeout, maybe was set to 60s?

rohan-varma commented:
> This change LGTM. Stamp to unblock
> 
> Curious, is it possible to reduce the response time of `NCCL_ASYNC_ERROR_HANDLING` for just this test? I would assume there is a timeout, maybe was set to 60s?

There is a way to change overall pg timeout by test name - https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L251. Will try this with ~5s PG timeout, as don't want to make the timeout too short to avoid flakiness.

rohan-varma committed:
Update on "[BE] Speed up runtime of test_ddp_model_diff_across_ranks"


As per https://github.com/pytorch/pytorch/issues/55583, this is the most expensive distributed test.

Instead of waiting for process 0 in this test to be taken down by
nccl_async_error_handling, just remove the barrier and let the process exit
when the backend is NCCL.

A slight downside here is that the test no longer verifies that the process
would be brought down by nccl_async_error_handling, but
nccl_async_error_handling is already well tested in other tests. If we feel we
need to ensure this for this test, then we can pass in a process group with a
smaller timeout as an alternative solution.

The test now runs in 4-6s as opposed to 70. Ran the test 1000 times to verify
no flakiness

Differential Revision: [D27672161](https://our.internmc.facebook.com/intern/diff/D27672161/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[BE] Speed up runtime of test_ddp_model_diff_across_ranks"


As per https://github.com/pytorch/pytorch/issues/55583, this is the most expensive distributed test.

Instead of waiting for process 0 in this test to be taken down by
nccl_async_error_handling, just remove the barrier and let the process exit
when the backend is NCCL.

A slight downside here is that the test no longer verifies that the process
would be brought down by nccl_async_error_handling, but
nccl_async_error_handling is already well tested in other tests. If we feel we
need to ensure this for this test, then we can pass in a process group with a
smaller timeout as an alternative solution.

The test now runs in 4-6s as opposed to 70. Ran the test 1000 times to verify
no flakiness

Differential Revision: [D27672161](https://our.internmc.facebook.com/intern/diff/D27672161/)

[ghstack-poisoned]

rohan-varma commented:
rocm error is unrelated
